.github/workflows/semantic-pull-request.yml

```diff
@@ -0,0 +1,116 @@
+# Linter to enforce semantic pull request titles (see https://www.conventionalcommits.org/)
+---
+name: 🔍 Semantic PR Validation
+
+on:
+  pull_request:
+    branches: 
+      - main
+    types:
+      - opened
+      - edited
+      - reopened
+      - ready_for_review
+
+permissions:
+  pull-requests: write
+
+jobs:
+  check_pr_title:
+    name: Check Pull Request Title
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run Semantic PR Validation
+        id: validation
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const prTitle = context.payload.pull_request.title;
+            const regex = /^(feat|fix|docs|style|refactor|perf|test|chore|build|ci|revert)(\([^)]+\))?(!?): .+/;
+
+            if (!regex.test(prTitle)) {
+              core.setFailed('Pull request title does not follow Conventional Commits specification. See PR comment for details.');
+            }
+      - name: Handle Invalid Title
+        if: failure()
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const commentMarker = '<!-- semantic-pr-comment -->';
+
+            const { data: comments } = await github.rest.issues.listComments({
```

Comment:
this will return issues and PRs.It will work, but consider replacing with 
github.rest.pulls.listReviewComments({
  owner: context.repo.owner,
  repo: context.repo.repo,
  pull_number: context.issue.number
});